test/Microsoft.ML.Tests/Transformers/TextNormalizer.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Runtime.Tools;
+using Microsoft.ML.Transforms.Text;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests.Transformers
+{
+    public class TextNormalizer : TestDataPipeBase
+    {
+        public TextNormalizer(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        private class TestClass
+        {
+            public string A;
+            [VectorType(2)]
+            public string[] B;
+        }
+
+        [Fact]
+        public void TextNormalizerWorkout()
+        {
+            var data = new[] { new TestClass() { A = "A 1, b. c! йЁ 24 ", B = new string[2] { "~``ё 52ds й vc", "6ksj94 vd ё dakl Юds Ё q й" } } };
+
+            var dataView = ComponentCreation.CreateDataView(Env, data);
+            var pipe = new TextNormalizerEstimator(Env, new[]{
+                    new TextNormalizerTransform.ColumnInfo("A", "NormA"),
+                    new TextNormalizerTransform.ColumnInfo("B", "NormB"),
+                });
+
+            TestEstimatorCore(pipe, dataView);
+            Done();
+        }
+
+        [Fact]
+        public void TestCommandLine()
+        {
+            Assert.Equal(Maml.Main(new[] { @"showschema loader=Text{col=A:TX:0} xf=TextNorm{col=B:A} in=f:\2.txt" }), (int)0);
+        }
+
+        [Fact]
+        public void TestOldSavingAndLoading()
```

Comment:
>TestOldSavingAndLoading [](start = 20, length = 23)

Both saving and loading seem to be done with the new transform code. Since you changed the versioninfo, it might be useful to have an other test where you load a model that was saved using the old transform code, so that we check that those models can still be loaded?

Reply:
Not anymore

---
In reply to: [226031938](https://github.com/dotnet/machinelearning/pull/1276#discussion_r226031938) [](ancestors = 226031938)